src/chat.rs

```diff
@@ -3702,11 +3702,11 @@ mod tests {
 
         // create group and sync it to the second device
         let a1_chat_id = create_group_chat(&a1, ProtectionStatus::Unprotected, "foo").await?;
-        a1.send_text(a1_chat_id, "ho!").await;
+        let sent = a1.send_text(a1_chat_id, "ho!").await;
         let a1_msg = a1.get_last_msg().await;
         let a1_chat = Chat::load_from_db(&a1, a1_chat_id).await?;
 
-        let a2_msg = a2.recv_msg(&a1.pop_sent_msg().await).await;
+        let a2_msg = a2.recv_msg(&sent).await;
```

Comment:
This test was wrong, `send_text()` already calls `pop_sent_msg()` internally. Because `pop_sent_msg` also was wrong and didn't actually delete the job since SMTP jobs were moved to the `smtp` table. I had to fix the latter:

```diff
-            .execute("DELETE FROM jobs WHERE id=?;", paramsv![rowid])
+            .execute("DELETE FROM smtp WHERE id=?;", paramsv![rowid])
```

because I needed the correct behavior for `test_handle_authres_fails()`, and so I also needed to fix this test here.